### PR TITLE
remove `is`, rewrite encodeFirestoreObject

### PR DESCRIPTION
**Pull request by bkendall**
### Description

* remove the `is` module, it's not necessary.
* rewrite encodeFirestoreObject to be TS
* new tests! huzzah!

### CODE REVIEW

File: src/test/firestore/encodeFirestoreValues.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import { expect } from "chai";
```
Comment (thechenky) [inline src/test/firestore/encodeFirestoreValues.ts]: This file's source file is called `encodeFirestoreValue.ts` but this one is plural, should change to make consistent.

File: src/firestore/encodeFirestoreValue.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
+import * as _ from "lodash";
+
+import * as FirebaseError from "../error";
+
+function isPlainObject(input: any): boolean {
+  return (
+    typeof input === "object" &&
+    input !== null &&
```
Comment (thechenky) [inline src/firestore/encodeFirestoreValue.ts]: should the null check go before checking the type?

File: package.json [hunk 3]
```diff
@@ -83,7 +83,6 @@
     "glob": "^7.1.2",
     "google-auto-auth": "^0.7.2",
     "inquirer": "^6.3.1",
-    "is": "^3.2.1",
```
[the remark this answers is not in the document]
Comment (bkendall) [reply package.json]: As far as I can tell, yes.

File: src/firestore/encodeFirestoreValue.ts (hunk 2 above)
Comment (bkendall) [reply src/firestore/encodeFirestoreValue.ts]: this is the order it was before... `typeof null` actually returns `"object"` (fun fact) but I don't think the order of this matters much.

File: src/test/firestore/encodeFirestoreValues.ts (hunk 1 above)
Comment (bkendall) [reply src/test/firestore/encodeFirestoreValues.ts]: Yeah, weird. I'll fix it.

**Pull request merged**